packages/manager/apps/pci-load-balancer/src/pages/create/steps/region/RegionStep.tsx

```diff
@@ -1,145 +1,187 @@
+import { useMemo } from 'react';
 import {
   OsdsLink,
+  OsdsMessage,
   OsdsSpinner,
   OsdsText,
 } from '@ovhcloud/ods-components/react';
-import { ODS_TEXT_LEVEL, ODS_TEXT_SIZE } from '@ovhcloud/ods-components';
+import {
+  ODS_ICON_NAME,
+  ODS_TEXT_LEVEL,
+  ODS_TEXT_SIZE,
+} from '@ovhcloud/ods-components';
 import { ODS_THEME_COLOR_INTENT } from '@ovhcloud/ods-common-theming';
-import { StepComponent } from '@ovh-ux/manager-react-components';
-import { useTranslation } from 'react-i18next';
-import { ShapesInputComponent } from '@ovh-ux/manager-pci-common';
-import { useEffect, useRef } from 'react';
+import {
+  StepComponent,
+  Links,
+  useProjectUrl,
+} from '@ovh-ux/manager-react-components';
+import { Trans, useTranslation } from 'react-i18next';
+import {
+  RegionSelector,
+  useProject,
+  usePCICommonContextFactory,
+  PCICommonContext,
+  TLocalisation,
+} from '@ovh-ux/manager-pci-common';
 import { TRegion } from '@/api/hook/useRegions';
 import { REGION_AVAILABILITY_LINK } from '@/constants';
 import { StepsEnum, useCreateStore } from '@/pages/create/store';
 import { useTracking } from '@/pages/create/hooks/useTracking';
-import { LabelComponent } from './components/Label.component';
-import { StackLabelComponent } from './components/StackLabel.component';
-import { GroupLabelComponent } from './components/GroupLabel.component';
-import { StackTitleComponent } from '@/pages/create/steps/region/components/StackTitle.component';
-import { useColumnsCount } from '@/pages/create/hooks/useColumnsCount';
 
 export type TRegionStepProps = {
   isLoading: boolean;
-  isMobile: boolean;
   regions: Map<string, TRegion[]>;
   ovhSubsidiary: string;
 };
+
+const isRegionWith3AZ = (regions: TRegion[]) =>
+  regions.some((region) => region.type === 'region-3-az');
+
 export const RegionStep = ({
   isLoading,
-  isMobile,
   regions,
   ovhSubsidiary,
 }: Readonly<TRegionStepProps>): JSX.Element => {
-  const { t: tCommon } = useTranslation('pci-common');
-  const { t: tCreate } = useTranslation('load-balancer/create');
-
-  const columnsCount = useColumnsCount();
+  const { t } = useTranslation(['load-balancer/create', 'pci-common']);
+  const { data: project } = useProject();
+  const projectUrl = useProjectUrl('public-cloud');
 
   const { trackStep } = useTracking();
 
   const store = useCreateStore();
 
-  // TODO: remove this once the class is added to step component directly in manager-react-components
-  useEffect(() => {
-    const refEl = document.getElementById('ref');
-    if (refEl) {
-      const stepEl = refEl.nextSibling;
-      if (stepEl) {
-        const targetEl = stepEl.childNodes.item(1);
+  const has3AZ = useMemo(() => {
+    const allRegions = regions ? [...regions.values()] : [];
```

Comment:
From what I see from the `props`, `regions` is always defined, so why do you use a ternary?

If it's to check if the `Map` is empty, calling `[...regions.values()]` already returns `[]`

```suggestion
    const allRegions = [...regions.values().flat()];
```

Reply:
Here the `type` of the `props` is not correct because `regions` is from an api and can be `undefined` until the call is finished. That's why I use the ternary

Reply:
@Tsiorifamonjena 
So, there is a big typing issue ;)
Please, make `regions` property as possibly `undefined`